android/java/org/chromium/chrome/browser/app/domain/PortfolioModel.java

```diff
@@ -150,28 +152,52 @@ public void fetchAssetsByType(TokenUtils.TokenType type, NetworkInfo selectedNet
-                                mPortfolioHelper = new PortfolioHelper(braveWalletBaseActivity,
-                                        mAllNetworkInfos, allAccounts.accounts);
-                                mPortfolioHelper.setSelectedNetworks(
-                                        NetworkUtils.nonTestNetwork(mAllNetworkInfos));
-                                mPortfolioHelper.fetchAssetsAndDetails(type, callback);
-                            });
-                        } else {
-                            mKeyringService.getAllAccounts(allAccounts -> {
-                                AccountInfo[] filteredAccounts =
+                        mKeyringService.getAllAccounts(allAccounts -> {
+                            AccountInfo[] filteredAccounts = allAccounts.accounts;
+                            List<NetworkInfo> selectedNetworks;
+                            if (selectedNetwork.chainId.equals(
+                                        NetworkUtils.getAllNetworkOption(mContext).chainId)) {
+                                selectedNetworks = NetworkUtils.nonTestNetwork(mAllNetworkInfos);
+                            } else {
+                                filteredAccounts =
                                         AssetUtils.filterAccountsByNetwork(allAccounts.accounts,
                                                 selectedNetwork.coin, selectedNetwork.chainId);
+                                selectedNetworks = Arrays.asList(selectedNetwork);
+                            }
 
-                                mPortfolioHelper = new PortfolioHelper(braveWalletBaseActivity,
-                                        mAllNetworkInfos, filteredAccounts);
-                                mPortfolioHelper.setSelectedNetworks(
-                                        Arrays.asList(selectedNetwork));
-                                mPortfolioHelper.fetchAssetsAndDetails(type, callback);
-                            });
-                        }
+                            mPortfolioHelper = new PortfolioHelper(
+                                    braveWalletBaseActivity, mAllNetworkInfos, filteredAccounts);
+                            mPortfolioHelper.setSelectedNetworks(selectedNetworks);
+                            mPortfolioHelper.fetchAssetsAndDetails(type, callback);
+                        });
+                    });
+        }
+    }
+
+    /**
+     * Fetch the assets of selected networks and filter by type.
+     * @param type by NFT, NON-NFT, ERC, SOL etc.
+     * @param selectedNetworks list to fetch assets from
+     * @param braveWalletBaseActivity instance for native services
+     * @param callback to get the @{code PortfolioHelper} object containing result
+     */
+    public void fetchAssets(TokenUtils.TokenType type, List<NetworkInfo> selectedNetworks,
+            BraveWalletBaseActivity braveWalletBaseActivity,
+            Callbacks.Callback1<PortfolioHelper> callback) {
+        synchronized (mLock) {
```

Comment:
I believe we don't need `synchronized`. LMK if I should remove this `synchronized` block.

Reply:
`fetchAssets` is always called from UI thread, so I feel that there's no need to make it `synchronized`. We can simply add `MainThread` annotation. And if we want to make it even more strict, include `ThreadUtils.assertOnUiThread()`.

Reply:
`ThreadUtils.assertOnUiThread()` is required for UI specific work so not needed here. Will remove it as suspected.